### DISCUSSION TITLE
Porting 10BASE-T1L driver for ADIN1110

### DISCUSSION THREAD

**Discussion by vChavezB** [zephyrproject-rtos/zephyr#52989] [Ideas] [answered]
I have been working with the ADIN1110 MAC-PHY chip from analog devices. The advantage of 10BASE-T1L is that you have a twisted pair connection that can also deliver PoDL over 1km. 

I have developed a [zephyr module](https://git.fh-aachen.de/iaam_embedded/zephyr/drivers/zephyr-adin1110/-/tree/main/) to use this MAC-PHY. However, I think it would be nice to integrate as an Ethernet driver to use it with `ip` stack of zephyr.

The main question is if this would be of interest and would be accepted in the project. Also I would like to mention that to use this MAC-PHY you need to use a driver provider by analog devices. This driver is composed of several source files and an include dir. What I have seen so far is that the PHY chips implemented in `drivers/net` are mostly of one source file and would not comply with the current project structure as I would need to add a folder inside `drivers/net` just for keeping a clean and ordered file structure (see how I have implemented as a module [here](https://git.fh-aachen.de/iaam_embedded/zephyr/drivers/zephyr-adin1110)).

Best regards
Victor

**Comment by oberstet** [comment, marked as answer]
fwiw, seems it'll land via this PR https://github.com/zephyrproject-rtos/zephyr/pull/60619 which is in review ..
